### PR TITLE
feat(scenarios): support report generation from multiple data sources

### PR DESCRIPTION
**Pull request by toidiu**
### Description of changes: 
https://github.com/aws/s2n-netbench/pull/3 added support for scenarios with multiple servers. However, the `report-tree` command was only including server.json and client.json files, which meant it was not possible to generate a report for multiple servers.

This PR modifies the report generation command, `report-tree`, to dynamically search the result directory and include all .json files.

### Testing:
CI report still works and I also tested this locally with multiple data sources.

Screenshot 2024-01-25 at 11 58 28 AM

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: netbench-cli/src/report_tree.rs [hunk 1]
```diff
@@ -33,31 +33,46 @@ impl ReportTree {
...
                     }};
                 }
 
-                push_scenario!(client_scenarios, "client");
-                push_scenario!(server_scenarios, "server");
+                // driver_path = sample/results/requst_response_incast/server-tcp
+                for json_result in driver_path.read_dir()? {
+                    let json_result_path = json_result?.path();
+
+                    let json_result_filename = json_result_path
```
Comment (toidiu) [inline netbench-cli/src/report_tree.rs:59]: only do this for .json

File: netbench-cli/src/report_tree.rs [hunk 2]
```diff
@@ -33,31 +33,46 @@ impl ReportTree {
...
-                                        self.out_dir
-                                            .join(scenario_name)
-                                            .join(concat!($name, "s.json")),
-                                    ),
-                                    ..Default::default()
-                                })
-                                .inputs
-                                .push(endpoint);
-                        }
+                    ($target:ident, $json_result_path:ident, $url_name:literal) => {{
```
Comment (toidiu) [inline netbench-cli/src/report_tree.rs:41]: see if its possible to get rid of the macro altogether

Comment (toidiu) [inline netbench-cli/src/report_tree.rs:41]: macro can be removed

File: netbench-cli/src/report_tree.rs [hunk 3]
```diff
@@ -33,31 +33,51 @@ impl ReportTree {
...
+
+                    // only process .json log files
+                    if !json_result_filename.ends_with(".json") {
+                        continue;
+                    }
+
+                    if json_result_filename.contains("server") {
+                        push_scenario!(server_scenarios, json_result_path, "servers.json");
+                    } else if json_result_filename.contains("client") {
+                        push_scenario!(client_scenarios, json_result_path, "clients.json");
```
Comment (toidiu) [inline netbench-cli/src/report_tree.rs:73]: Even though the macro is much simpler now, it feels like there is enough de-dupliation of logic to warrant the use of a macro.

**Review by camshaft — DISMISSED** [review]

**Review by camshaft — APPROVED** [review]

**Pull request merged**